FiftyOne.Pipeline.Core/FlowElements/Pipeline.cs

```diff
@@ -239,7 +265,7 @@ public IReadOnlyList<IFlowElement> FlowElements
             _elementsByType = new Dictionary<Type, List<IFlowElement>>();
             AddElementsByType(_flowElements);
 
-            _elementAvailableProperties = GetElementAvailableProperties(_flowElements);
+            _ = ElementAvailableProperties; // perform caching attempt (default happy path)
```

Comment:
do we really need this?  if we are lazifying, let's lazify!

Reply:
Yes, it indirectly invokes pipeline consistency checks, see [[CloudAspectEngineBase.GetInstance](https://github.com/51Degrees/pipeline-dotnet/blob/main/FiftyOne.Pipeline.CloudRequestEngine/FlowElements/CloudAspectEngineBase.cs#L100)] and [[CloudAspectEngineBaseTests.CloudEngines_NoRequestEngine](https://github.com/51Degrees/pipeline-dotnet/blob/main/Tests/FiftyOne.Pipeline.CloudRequestEngine.Tests/CloudAspectEngineBaseTests.cs#L585)].

Reply:
then maybe let's add a comment that explains exactly this motivation, rather than a self-referential comment that we have now :)

Reply:
Added: https://github.com/51Degrees/pipeline-dotnet/pull/46/commits/a77d3453d333eaae5b5e49e677bbfd7d2584a8bc